packages/interop-tests/src/__test__/interop.test.ts

```diff
@@ -94,17 +89,18 @@ beforeEach(async () => {
 
   browserWallet.onSendMessage(message => {
     if (isJsonRpcNotification(message)) {
+      // TODO: Since we're not awaiting this it can execute while knex is being destroyed
```

Comment:
Why not await it?

Reply:
The problem is that we're not `await`ing pushMessage so we're not waiting for the  `pushMessage` to complete.

So the browser wallet can try and send a message **before** the test completes (and the server wallet is destroyed) that continues executing **after** the test completes.

This is because the `engine.pushMessage` is triggered in the listener. This means the test can't be smart enough to `await` the completion of `pushMessage` before allowing things to be destroyed.

Reply:
Is there a test case where we want the test to finish before the messaging has finished for some protocol?

I'm sure there could be, but my gut says that all messages should be resolved before the test is finished.